src/Indexer.ts

```diff
@@ -0,0 +1,73 @@
+import * as path from 'path'
+import * as url from 'url'
+
+import * as ts from 'typescript'
+
+import { Input } from './Input'
+import * as lsif from './lsif'
+import { LsifSymbol } from './LsifSymbol'
+import { Options, lsif_typed } from './main'
+import { Packages } from './Packages'
+import { Visitor } from './Visitor'
+
+export class Indexer {
+  public options: Options
+  public program: ts.Program
+  public checker: ts.TypeChecker
+  public symbolsCache: Map<ts.Node, LsifSymbol> = new Map()
+  public packages: Packages
+  constructor(public readonly config: ts.ParsedCommandLine, options: Options) {
+    this.options = options
+    this.program = ts.createProgram(config.fileNames, config.options)
+    this.checker = this.program.getTypeChecker()
+    this.packages = new Packages(options.project)
+  }
+  public index(): void {
+    this.options.writeIndex(
+      new lsif_typed.Index({
+        metadata: new lsif_typed.Metadata({
+          project_root: url.pathToFileURL(this.options.projectRoot).toString(),
+          tool_info: new lsif_typed.ToolInfo({
+            name: 'lsif-node',
```

Comment:
Should we use `lsif-typescript` here?

Reply:
And maybe we should bump the version down to `0.1.0` to indicate that this isn't stable yet?